### PR TITLE
Streams bug fixes

### PR DESCRIPTION
**Pull request by torosent**

### CODE REVIEW

File: pkg/stream/eventhubs/eventhubs.go [hunk 1]
```diff
@@ -76,14 +78,23 @@ func NewConsumer(ctx context.Context, eventHubConnStr string, storageAccountName
...
 func (e *eventhubs) Receive(receiveFunc stream.ReceiveFunc) error {
 
-	e.eph.StartNonBlocking(e.ctx)
 	e.receiveFunc = receiveFunc
 	_, err := e.eph.RegisterHandler(e.ctx, e.handleEvent)
+	if err != nil {
+		return err
+	}
+
+	err = e.eph.Start(e.ctx)
```
Comment (Ilanak) [inline pkg/stream/eventhubs/eventhubs.go]: you can just return

File: pkg/stream/eventhubs/eventhubs.go [hunk 2]
```diff
@@ -76,14 +78,23 @@ func NewConsumer(ctx context.Context, eventHubConnStr string, storageAccountName
...
 	return err
 }
 
 func (e *eventhubs) handleEvent(ctx context.Context, event *eh.Event) error {
-	err := e.receiveFunc(ctx, *event.PartitionKey, event.ID, string(event.Data))
+	key := "-1"
+	if event.PartitionKey != nil {
+		key = *event.PartitionKey
+	}
+	err := e.receiveFunc(ctx, key, event.ID, string(event.Data))
```
Comment (Ilanak) [inline pkg/stream/eventhubs/eventhubs.go]: same

File: pkg/stream/eventhubs/eventhubs.go [hunk 3]
```diff
@@ -76,21 +78,26 @@ func NewConsumer(ctx context.Context, eventHubConnStr string, storageAccountName
 //Receive message from eventhub partition.
 func (e *eventhubs) Receive(receiveFunc stream.ReceiveFunc) error {
 
-	e.eph.StartNonBlocking(e.ctx)
```
Comment (eladiw) [inline pkg/stream/eventhubs/eventhubs.go]: was the core issue was starting the eph before setting the receive func?

File: docs/tutorial_scheduler.md [hunk 4]
```diff
@@ -210,20 +210,20 @@ odbc:server=<serverName>.database.windows.net;user id=<userId>;password=<passwor
...
+
 ```json
   "streamConfig": {
     "ehConfig": {
-      "ehNamespace": "<ehNamespace>",
-      "ehName": "<ehName>",
-      "ehConnectionString": "<ehConnectionString>",
-      "ehKeyName": "<ehKeyName>",
-      "ehKeyValue": "<ehKeyValue>"
+      "ehConnectionString": "<ehConnectionString>", // EH connection string. It is recommended to generate a connection string from EH and NOT from EH namespace.
```
[the remark this answers is not in the document]
Comment (eladiw) [reply docs/tutorial_scheduler.md]: so the end result, the conn string will look the same anyway.
saying recommended means you have an option what to supply here...
consider rephrasing. minor...

how about: "EH connection string, the easiest option is to generate from the EH and not from the EH namespace"

**Comment by eladiw** [comment]
LGTM. minor comment on rephrasing one comment.

**Pull request merged**